config/application.rb

```diff
@@ -77,6 +77,15 @@ class Application < Rails::Application
 
     config.middleware.insert_before "ActionDispatch::DebugExceptions", "LogstasherCatchAllErrors"
     config.middleware.use Rack::MobileDetect
+
+    config.middleware.insert_before 0, "Rack::Cors" do
```

Comment:
I guess I'll start with saying that it *does* work. I've tested running without it and it definitely doesn't work so it's not as if there's some other config that's actually doing the heavy lifting.

I'm under the impression that, as it's a CORS module, `OPTIONS` is mandatory. How can it work otherwise? :confused: I was sure I'd read somewhere that what you're configuring here is not "what methods are allowed for this endpoint" but instead "what methods should the CORS response *say* are allowed by the client". Now I'm looking for where I read that, I can't find it, but this does seem to be how it works.

I know the examples talk about explicitly adding `OPTIONS` to the list, and I've tested with it and it makes no difference. **If you'd prefer that I add `:options` to the list, I can**, but I think how it stands captures the intent that the CORS response should indicate that you can only do a `POST`.

As for adding `rack-cors` as a dependency, I did that because 1. I'm a Ruby n00b, and 2. I was following the suggestion [from the docs](https://github.com/doorkeeper-gem/doorkeeper/wiki/%5BCORS%5D-Cross-Origin-Resource-Sharing). I should've done a better search how you'd already done CORS in the project. **If you want me to drop the dependency and swap to the existing method, I can do so.**

Reply:
Leave the `rack-cors` gem in there, just talked with @pleary and he's fine with it. No need to be explicit about `OPTIONS` if it's working for you.